libclc/cmake/modules/AddLibclc.cmake

```diff
@@ -90,9 +90,20 @@ function(link_bc)
     ${ARGN}
   )
 
+  set( LINK_INPUT_ARG ${ARG_INPUTS} )
+  if( WIN32 OR CYGWIN )
+    # Create a response file in case the number of inputs exceeds command-line
+    # character limits on certain platforms.
+    file( TO_CMAKE_PATH ${LIBCLC_ARCH_OBJFILE_DIR}/${ARG_TARGET}.rsp RSP_FILE )
+    # Turn it into a space-separate list of input files
+    list( JOIN ARG_INPUTS " " RSP_INPUT )
+    file( WRITE ${RSP_FILE} ${RSP_INPUT} )
+    set( LINK_INPUT_ARG "@${RSP_FILE}" )
+  endif()
+
   add_custom_command(
     OUTPUT ${ARG_TARGET}.bc
-    COMMAND libclc::llvm-link -o ${ARG_TARGET}.bc ${ARG_INPUTS}
+    COMMAND libclc::llvm-link -o ${ARG_TARGET}.bc ${LINK_INPUT_ARG}
```

Comment:
I suspect this will break the dependencies for re-building `libclc` after changes, did you try it?

We can go with this to fix the build but we should follow up on that.

Reply:
The dependencies should still be on each individual argument as they're provided to the `DEPENDS` argument. But I'm not sure what would happen if you delete or alter the `.rsp` file itself. Is that what you're asking?

Reply:
Oh I missed the `DEPENDS`, that's probably fine then, I'm always wary of custom commands dependencies

Reply:
Haha well rightly so! I deleted the `.rsp` locally and CMake doesn't know it has to re-run to regenerate that file. So this does need a little more time in the oven.

Reply:
Pushed another commit which should regenerate the `.rsp` file if it's deleted, by re-running cmake. It's a worst-case scenario so shouldn't really happen in practice.